src/main/java/mytown/util/TownBlockKey.java

```diff
@@ -0,0 +1,25 @@
+package mytown.util;
+
+public class TownBlockKey {
+    int dim, x, z;
+    public TownBlockKey(int dim, int x, int z) {
+        this.dim = dim;
+        this.x = x;
+        this.z = z;
+    }
+
+    @Override
+    public int hashCode() {
+        return this.dim ^ this.x ^ this.z;
```

Comment:
Since dim will most likely always be 0, id rather do something like

``return (17 + this.dim) ^ this.x ^ this.z;``

Reply:
i don't get why you would add 17. Gonna test the pullrequest on a minimal working environment to see if it doesn't break existing towns.

Reply:
Since this is using xor, would that make any difference?

Reply:
if you have not evenly distributed numbers, you want to add a prime number to them, to have a more evenly spread out hash.